### PR TITLE
#1718 - Deprecation Warning

### PR DESCRIPTION
**Pull request by JohnPTobe**
##### Checklist

- [x] `manage.py test` passes
- [x] tests are included
- [x] documentation is changed or added

### Affected app(s)
- job
- recipe

### Description of change
Added fields to job-type details and recipe-type details apis to show which recipes each belongs to so they can be easily found if they need to be edited.
Automatically deprecate parent recipes when jobs/recipes are deprecated regardless of auto-update flag. Parent recipes aren't automatically reactivated.
Changed job/recipe patch endpoints to return 200 OK response with validation dict for warnings.

### CODE REVIEW

**Comment by JohnPTobe** [comment]
It's needs tests and documentation yet but I wanted @emimaesmith and @Rachel-ais to look at this to see if you agree this would work for warning the user about deprecated recipes.  Also, could we use this to display recipe types a job/recipe is part of on the details page?

**Comment by JohnPTobe** [comment]
Ready for review

File: scale/job/views.py [hunk 1]
```diff
@@ -363,6 +369,7 @@ def patch(self, request, name, version):
         except JobType.DoesNotExist:
             raise Http404
 
+        jt = None
```
Comment (emimaesmith) [inline scale/job/views.py]: Leftover debugging variable?

File: scale/recipe/models.py [hunk 2]
```diff
@@ -1236,8 +1243,13 @@ def edit_recipe_type_v6(self, recipe_type_id, title=None, description=None, defi
         
         if is_active is not None:
             recipe_type.is_active = is_active
-            if auto_update:
+            if is_active == False:
```
Comment (emimaesmith) [inline scale/recipe/models.py]: Same question as above about removing the auto_update.

File: scale/recipe/views.py [hunk 3]
```diff
@@ -233,6 +233,7 @@ def patch_v6(self, request, name):
         except RecipeType.DoesNotExist:
             raise Http404
 
+        rt = None
```
Comment (emimaesmith) [inline scale/recipe/views.py]: Leftover debugging variable?

File: scale/job/models.py [hunk 4]
```diff
@@ -2078,8 +2082,13 @@ def edit_job_type_v6(self, job_type_id, manifest=None, docker_image=None, icon_c
         if is_active is not None and job_type.is_active != is_active:
             job_type.deprecated = None if is_active else timezone.now()
             job_type.is_active = is_active
-            if auto_update:
+            if is_active == False:
```
[the remark this answers is not in the document]
Comment (JohnPTobe) [reply scale/job/models.py]: After thinking a bit about it, I think it makes more sense to deprecate all parent recipes when a child is deprecated as they can't be run until the child is fixed.  The auto_update flag should only handle automatically updating the revision of a child job/recipe when it won't break anything.  Also, we obviously don't want to automatically re-enable recipes as there could be other things causing a recipe to be deprecated and a user should manually review a recipe before it's activated.

File: scale/job/views.py (hunk 1 above)
Comment (JohnPTobe) [reply scale/job/views.py]: yep, missed a spot

File: scale/recipe/views.py (hunk 3 above)
Comment (JohnPTobe) [reply scale/recipe/views.py]: another spot

**Comment by JohnPTobe** [comment]
Ready for re-review. If you're unsure about the auto_update flag we can discuss it more.

**Pull request merged**